### ISSUE TITLE
Superclass must be a class

### ISSUE THREAD

**Issue by davingee** [spree-contrib/spree_paypal_express#26]
Hello,
Just installed spree into rails 3.1.1 app got it running then added spree_paypal express to the gemfile bundle installed then   rake spree_paypal_express:install. but it failed it copied the db migration over and that was it then tried to rake db:migrate and got -----

/Users/scott/.rvm/gems/ruby-1.8.7-p352/bundler/gems/spree_paypal_express-b597130b5ef3/app/models/spree/billing_integration/paypal_express.rb:1: superclass must be a Class (Module given) (TypeError)
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:456:in `load'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:456:in`load_file'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:640:in `new_constants_in'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:455:in`load_file'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:342:in `require_or_load'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:489:in`load_missing_constant'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:181:in `const_missing'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:179:in`each'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:179:in `const_missing'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/bundler/gems/spree_paypal_express-b597130b5ef3/lib/spree_paypal_express/engine.rb:20
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/lazy_load_hooks.rb:34:in`call'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/lazy_load_hooks.rb:34:in `execute_hook'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/lazy_load_hooks.rb:43:in`run_load_hooks'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/lazy_load_hooks.rb:42:in `each'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/lazy_load_hooks.rb:42:in`run_load_hooks'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/application/finisher.rb:56
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/initializable.rb:30:in `instance_exec'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/initializable.rb:30:in`run'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/initializable.rb:55:in `run_initializers'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/initializable.rb:54:in`each'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/initializable.rb:54:in `run_initializers'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/application.rb:96:in`initialize!'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/railtie/configurable.rb:30:in `send'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/railties-3.1.1/lib/rails/railtie/configurable.rb:30:in`method_missing'
    from /Users/scott/new_rails/site5/prettylulu.com/config/environment.rb:5
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:240:in `require'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:240:in`require'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:223:in `load_dependency'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:640:in`new_constants_in'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:223:in `load_dependency'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/activesupport-3.1.1/lib/active_support/dependencies.rb:240:in`require'
    from /Users/scott/new_rails/site5/prettylulu.com/config.ru:4
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/rack-1.3.5/lib/rack/builder.rb:51:in `instance_eval'
    from /Users/scott/.rvm/gems/ruby-1.8.7-p352/gems/rack-1.3.5/lib/rack/builder.rb:51:in`initialize'
    from /Users/scott/new_rails/site5/prettylulu.com/config.ru:1:in `new'
    from /Users/scott/new_rails/site5/prettylulu.com/config.ru:1

I tried to start it the app and got the same thing

**Comment by railskipl** [comment]
Any luck for above issue  ?

**Comment by davingee** [comment]
There were a lot of issues with spree 7 (with what i was doing) so I went with spree 6 and it was perfect; well it worked a bit friendlier. However I came to the conclusion that it was loading the paypal-spree gem first and the class it was looking for "billing_integration" was not yet initialized. the gem was getting loaded into bundler/gems instead of the gems directory. I tried packing them into vendor/bundler but it still split the spree extensions into the bundler/gems directory. I had it working when all the spree gems and the spree extensions were all being loaded into the bundler/gems folder.  So I am sorry I was on a deadline and went with 6.0. the versionfile of the paypal_spree gem states the version for which spree hope that helps cheers
